docs/src/custom.md

```diff
@@ -0,0 +1,126 @@
+- Your transform may be invertible in the mathematical sense
+
+!!! note "Revertible does not imply invertible"
+    Reverting assumes that the transform has been applied already and reverts it. Meanwhile, invertible implies that there is a one-to-one mapping between the input and output tables so a table can be inverted to a corresponding input to the transform even if it was not transformed a priori.
+
+## Creating a custom transform
+In the following we shall demonstrate the steps to apply a custom transform. 
+
+### 1. Declare a new type for your transform
+The declaration struct should subtype `TransformsBase.Transforms` and it should have a named field for each input needed to apply the transform besides of the main table. For instance, if you want to call your transform `Standardize` and it takes two boolean inputs `center` and `scale`, then you should declare:
+
+```julia
+struct Standardizer <: TransformsBase.Transform
+    center::Bool
+    scale::Bool
+end
+```
+You may implement keyword instructors as needed if some of the hyperparameters are optional. Thus, we follow up with:
+```julia
+Standardize(;center::Bool=true, scale::Bool=true) = Standardize(center, scale)
+```
+
+### 2. Implement the `apply` method for your transform
+The `apply` method takes an object of your transform type and a table and returns a new table and cache. Suppose that the `Standardize` transform should zero-mean each column if `center` is true and scale each column to unit variance if `scale` is true, then the `apply` method should be implemented as follows:
+
+```julia
+using Statistics
+
+function TransformsBase.apply(transform::Standardize, X)
+    # convert the table to a matrix
+    Xm = Tables.matrix(X)
+    # compute the means and stds
+    μ = transform.center ? mean(Xm, dims=1) : zeros(1, size(Xm, 2))
+    σ = transform.scale ? std(Xm, dims=1) : ones(1, size(Xm, 2))
+    # standardize the data
+    Xm = (Xm .- μ) ./ σ
+    # convert back to table
+    Xm = X |> Tables.materializer(Xm)
+    # return the table and cache that may help reapply or revert later
+    return Xm, (μ, σ)
+end
+```
+
+That's it really! Your transform now behaves like any table transform. For instance, we can do
+```julia
+using TableTransforms
+
+X = (A=[1, 2, 3], B=[4, 5, 6])
+Xt = X |> Standardize() |> Identity() |> Select([:A])
+```
+
+It holds, however, that in case your transform can be reapplied, is revertible, or is invertible then you should continue implementing the interface to support such functionality.
+
+### 3. Optionally implement `reapply`
+We need this in case of the `Standarize` transform because after computing the mean and std for some train table we may want to apply the transform directly given a test table. Hence, we implement `reapply` which has the same signature as apply but it takes an extra argument for the cache and doesn't return it.
```

Comment:
```suggestion
### 3. Optionally implement `reapply`

We need this in case of the `Standarize` transform because after computing the mean and std for some train table we may want to apply the transform directly given a test table. Hence, we implement `reapply` which has the same signature as apply but it takes an extra argument for the cache and doesn't return it.

```